### DISCUSSION TITLE
Question about "Isolate: It looks like you've created the database class multiple times" message.

### DISCUSSION THREAD

**Discussion by vicenterusso** [simolus3/drift#1782] [Q&A]
Hello!

This is my scenario:

I have `Timer.periodic` started after login to continuous check database records and send to API (or retrieve). I want it to be on isolates.

Apparently it's working, data is being recorded and timer is being persisted through screens. My question is about the error above.

I'm creating 2 databases, standard way and isolate way. After I create isolate database, can I use it normally through the whole app, and even register it with services locator (like Get It), for example?

**Comment by simolus3** [comment]
If you're seeing that message, you're instantiating the same database class, multiple times, on the _same isolate_. A `DriftIsolate` will internally run queries on a background isolate, but that doesn't necessarily mean that your code using a `DriftIsolate` is running on a background isolate too.

If you want the whole synchronization of records to happen on a background isolate, use something like `compute` in the callback and just create a regular instance of your app's database in the compute callback running on a background isolate.

I don't see much use of creating two databases, one with a `DriftIsolate` and one without. I'd probably just use one single database with a `DriftIsolate` instead.

> After I create isolate database, can I use it normally through the whole app, and even register it with services locator (like Get It), for example?

Yes! A `DriftIsolate` is an implementation-level thing to internally run queries on a background isolate. This isolate is managed by drift, all your code using the database is still running on the main isolate. So you can use the database just like you would use or register any other object.

**Reply by vicenterusso** [reply]
Perfect, Tks!
